yb-voyager/src/queryparser/helpers.go

```diff
@@ -105,6 +118,118 @@ func GetStringValueFromNode(nodeMsg protoreflect.Message) string {
+	strField := node.Descriptor().Fields().ByName(protoreflect.Name(fieldName))
+	if strField == nil || !node.Has(strField) {
+		return ""
+	}
+	return node.Get(strField).String()
+}
+
+// extractAConstString extracts the string from an 'A_Const' node's 'sval' field
+// Sample example:: rowexpr:{a_const:{sval:{sval:"//Product"}  location:124}}
+func extractAConstString(aConstMsg protoreflect.Message) string {
+	// Extract the 'sval' field from 'A_Const'
+	svalField := aConstMsg.Descriptor().Fields().ByName("sval")
+	if svalField == nil || !aConstMsg.Has(svalField) {
+		return ""
+	}
+
+	svalMsg := aConstMsg.Get(svalField).Message()
+	if svalMsg == nil || !svalMsg.IsValid() {
+		return ""
+	}
+
+	// Ensure svalMsg is of type 'pg_query.String'
+	if svalMsg.Descriptor().FullName() != "pg_query.String" {
+		return ""
+	}
+
+	// Extract the actual string value from 'pg_query.String'
+	return extractStringField(svalMsg, "sval")
+}
+
+// traverseAndExtractAConst traverses to a specified field and extracts the 'A_Const' string value
+// Sample example:: rowexpr:{type_cast:{arg:{a_const:{sval:{sval:"/order/item"}
+func traverseAndExtractAConst(nodeMsg protoreflect.Message, fieldName string) string {
+	field := nodeMsg.Descriptor().Fields().ByName(protoreflect.Name(fieldName))
+	if field == nil || !nodeMsg.Has(field) {
+		return ""
+	}
+
+	childMsg := nodeMsg.Get(field).Message()
+	if childMsg == nil || !childMsg.IsValid() {
+		return ""
+	}
+
+	return GetStringValueFromNode(childMsg)
+}
+
+/*
+isXPathExprForXmlTable checks whether a given string is a valid XPath expression for XMLTABLE()'s rowexpr.
+It returns true if the expression starts with '/' or '//', indicating an absolute or anywhere path.
+This covers the primary cases used in XMLTABLE() for selecting XML nodes as rows.
+
+XPath Expression Cases Covered for XMLTABLE():
+1. Absolute Paths:
+- Starts with a single '/' indicating the root node.
+- Example: "/library/book"
+
+2. Anywhere Paths:
+- Starts with double '//' indicating selection from anywhere in the document.
+- Example: "//book/author"
+
+For a comprehensive overview of XPath expressions, refer to:
+https://developer.mozilla.org/en-US/docs/Web/XPath
+*/
+func IsXPathExprForXmlTable(expression string) bool {
+	// Trim leading and trailing whitespace
+	expression = strings.TrimSpace(expression)
+	if expression == "" {
+		return false
+	}
+
+	// Check if the expression starts with '/' or '//'
+	return strings.HasPrefix(expression, "/") || strings.HasPrefix(expression, "//")
+}
+
 func GetMsgFullName(msg protoreflect.Message) string {
 	return string(msg.Descriptor().FullName())
 }
+
+// IsParameterPlaceholder checks if the given node represents a parameter placeholder like $1
+func IsParameterPlaceholder(nodeMsg protoreflect.Message) bool {
+	if nodeMsg == nil || !nodeMsg.IsValid() {
+		return false
+	}
+
+	// Retrieve the 'node' oneof descriptor
+	nodeOneof := nodeMsg.Descriptor().Oneofs().ByName("node")
```

Comment:
Consider taking this logic (getting nodeField out of nodeMsg) out to a different function if it's being done in multiple places.

Reply:
done